pandas-stubs/core/frame.pyi

```diff
@@ -633,37 +633,6 @@ class DataFrame(NDFrame, OpsMixin):
         axis: AxisType = ...,
         fill_value: Hashable | None = ...,
     ) -> DataFrame: ...
-    @overload
-    def set_index(
```

Comment:
I think this should wait until 1.5 is released. 1.5 deprecates `inplace` and adds the new `copy` argument.

Might be worth writing a `@skip_on_nightly` decorator for the tests.

Reply:
Other option is we start maintaining a fork related to 1.5 changes.

Reply:
I reverted this since inplace is back for 1.5.0